TemplateBicep/main.bicep

```diff
@@ -0,0 +1,66 @@
+param location string = resourceGroup().location
+param uniqueSolutionPrefix string
+param discoveryZipUrl string
+
+module iotHub 'modules/iothub.bicep' = {
+  name: 'iotHub'
+  params: {
+    name: '${uniqueSolutionPrefix}hub'
+    location: location
+  }
+}
+
+module storage 'modules/storage.bicep' = {
+  name: 'storage'
+  params: {
+    storageAccountName: '${uniqueSolutionPrefix}storage'
+    storageAccountType: 'StorageV2'
+    credentialsContainerName: 'stationcredentials'
+    firmwareUpgradesContainerName: 'fwupgrades'
+    location: location
+  }
+}
+
+module redisCache 'modules/redis.bicep' = {
+  name: 'redisCache'
+  params: {
+    uniqueSolutionPrefix: uniqueSolutionPrefix
+    location: location
+  }
+}
+
+module function 'modules/function.bicep' = {
+  name: 'function'
+  params: {
+    appInsightName: observability.outputs.appInsightName
+    logAnalyticsName: observability.outputs.logAnalyticsName
+    deployDevice: true
```

Comment:
deployDevice useAzureMonitorOnEdge Should be user defined params

Reply:
Changes added. Please review.